### PR TITLE
Bump dkregistry and detect when authentication without credentials is required when scraping a registry

### PR DESCRIPTION
**Pull request by steveej**
See https://github.com/camallo/dkregistry-rs/pull/168 for the upstream change and the problem description.

- [x] Requires https://github.com/camallo/dkregistry-rs/pull/180
- [x] Requires https://github.com/camallo/dkregistry-rs/pull/181

Testing the scraper plugin against the OpenShift internal registry revealed that it requires anonymous authentication as well, so this will be needed for the effort of moving our test images there.

### CODE REVIEW

File: cincinnati/src/plugins/internal/graph_builder/release_scrape_dockerv2/registry/mod.rs
Comment (vrutkovs) [inline cincinnati/src/plugins/internal/graph_builder/release_scrape_dockerv2/registry/mod.rs]: Which username and password are sent when `force_auth` is true?

Comment (steveej) [reply cincinnati/src/plugins/internal/graph_builder/release_scrape_dockerv2/registry/mod.rs]: The ones which are set, which could be none.

Comment (vrutkovs) [reply cincinnati/src/plugins/internal/graph_builder/release_scrape_dockerv2/registry/mod.rs]: In that case why can't `None` be sent right away, trap 401 and try without username/password?

Comment (steveej) [reply cincinnati/src/plugins/internal/graph_builder/release_scrape_dockerv2/registry/mod.rs]: I'll try to fill in your question:
In the case of username and password being `None`, we try to fetch the releases without authentication first, and if it tails with a 401 error code we try to get a token anonymously and then retry the release fetching.
I think that can be done. I have just kept it simpler and left the decision for anonymous authentication to the configuration.

Comment (steveej) [reply cincinnati/src/plugins/internal/graph_builder/release_scrape_dockerv2/registry/mod.rs]: To drive this conversation towards resolve, @vrutkovs are you asking for a change in the above described manner or were you merely curious?

**Review by vrutkovs — CHANGES_REQUESTED** [review]
anonymous_auth is not necessary, it can be derived based username/password and registry responses

**Comment by steveej** [comment]
I want to land some dkregistry-rs PRs first which are now linked in the first comment.

/hold

**Comment by steveej** [comment]
The dkregistry-rs PRs have landed and this should be good to go when CI is green.

/hold

**Comment by steveej** [comment]
I think this timing issue could be caused by high load on the CI node:

```
 thread 'plugins::tests::process_blocking_succeeds' panicked at 'took 2.139688319s despite timeout of 2s', cincinnati/src/plugins/mod.rs:675:9 
```

/retest

**Review by vrutkovs — APPROVED** [review]
/lgtm

**Pull request merged**
